### PR TITLE
Implement Verb ACL

### PR DESCRIPTION
**Pull request by AymericDu**

### CODE REVIEW

File: conf/default.cfg [hunk 1]
```diff
@@ -255,3 +255,7 @@ use = egg:swift3#swift3
 # OpenIO version
 use = egg:oioswift#copy
 object_post_as_copy = false
+
+[filter:verb_acl]
+use = egg:oioswift#verb_acl
+verb_acl = PUT:127.0.0.1,127.0.0.2;DELETE:127.0.0.3,127.0.0.4;PUT,DELETE:127.0.0.4,127.0.0.6
```
Comment (murlock) [inline conf/default.cfg]: Do you accept `192.168.1.0/24` ?

File: oioswift/common/middleware/verb_acl.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        self.logger.debug("Verb ACL: " + str(self.verb_acl))
+
+    def __call__(self, env, start_response):
+        req = Request(env)
+        if req.method in self.verb_acl:
+            remote = get_remote_client(req)
+            matched = False
+            for block in self.verb_acl[req.method]:
+                if remote.startswith(block):
+                    matched = True
```
Comment (murlock) [inline oioswift/common/middleware/verb_acl.py]: could be rewritten as
```
for block in ...
    if remote....
        break
else:
   raise HTTP...

File: oioswift/common/middleware/verb_acl.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    """
+
+    def __init__(self, app, conf):
+        self.app = app
+        self.logger = get_logger(conf)
+
+        self.verb_acl = {}
+        verb_acl = conf.get('verb_acl', None)
+        if not verb_acl:
+            raise ValueError('verb_acl: Not initialized in the config file.')
```
Comment (murlock) [inline oioswift/common/middleware/verb_acl.py]: Minor: position is not checked. We should provide an helper to check position, at least, please add a `TODO`

File: oioswift/common/middleware/verb_acl.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                    self.verb_acl.setdefault(method.upper(), []).append(block)
+        self.logger.debug("Verb ACL: " + str(self.verb_acl))
+
+    def __call__(self, env, start_response):
+        req = Request(env)
+        if req.method in self.verb_acl:
+            remote = get_remote_client(req)
+            matched = False
+            for block in self.verb_acl[req.method]:
+                if remote.startswith(block):
```
Comment (murlock) [inline oioswift/common/middleware/verb_acl.py]: If verb_acl contains `DELETE:192,192.168.100.1`, every request from 192 will be accepted, it is intended ?

File: conf/default.cfg (hunk 1 above)
Comment (AymericDu) [reply conf/default.cfg]: Not yet.

File: oioswift/common/middleware/verb_acl.py (hunk 4 above)
Comment (AymericDu) [reply oioswift/common/middleware/verb_acl.py]: Yes, it's wanted.

File: conf/default.cfg (hunk 1 above)
Comment (fvennetier) [reply conf/default.cfg]: Is that a whitelist or a blacklist? Without reading the code it is not obvious.

**Pull request merged**